modules/pam_usertype/pam_usertype.c

```diff
@@ -121,6 +121,7 @@ pam_usertype_get_uid(struct pam_usertype_opts *opts,
         }
 
         *_uid = pwd->pw_uid;
+        free(pwd);
```

Comment:
I think this change introduces a double-free.
Why do you think the memory returned by pam_modutil_getpw* functions should be freed explicitly?

Reply:
I misunderstood the documentation. So, I've checked it with a simple example and yes, this would mean a double free. I'll remove it when I update the patches.